### DISCUSSION TITLE
Is react-freeze enabled by default in react-navigation v6?

### DISCUSSION THREAD

**Discussion by fabiendem** [react-navigation/react-navigation#11241] [Q&A] [answered]
Hi all,

I understand that `react-navigation` v6 uses `react-native-screens` which uses `react-freeze`.

Reading the doc at https://reactnavigation.org/docs/stack-navigator/#detachinactivescreens it's not clear to me if we still need to enable the `react-freeze` integration by calling `enableFreeze(true)` at the app level from the `react-native-screens` library.

When at https://reactnavigation.org/docs/stack-navigator/#detachinactivescreens we say 
> This enables integration with [react-native-screens](https://github.com/software-mansion/react-native-screens).

Does it mean it just enables the integration but doesn't activate the feature?

I am trying to understand what's the minimal setup for react-freeze to run properly with react-navigation in our app.

Cheers !

**Comment by satya164** [comment, marked as answer]
`react-native-screens` integration refers to the detaching the inactive screens to reduce memory consumption. React Freeze is not enabled by default in React Native screens. You need to enable that explicitly if you want https://github.com/software-mansion/react-native-screens#experimental-support-for-react-freeze

**Reply by fabiendem** [reply]
OK thanks, so can you confirm that for [`detachInactiveScreens`](https://reactnavigation.org/docs/stack-navigator/#detachinactivescreens) to have any effect, we must enable `react-freeze` using the instructions at https://github.com/software-mansion/react-native-screens#experimental-support-for-react-freeze ?

(that's maybe what you meant but I am not sure how to interpret it)

**Reply by satya164** [reply]
No, React Freeze is just an additional functionality provided by react-native-screens. The main functionality is to save resources by detaching the native view for inactive screens which is integrated by default with `detachInactiveScreens`. Freezing the inactive screens is an additional experimental feature that needs to be enabled separately.

**Reply by fabiendem** [reply]
Thank you, that's super clear now.
